### PR TITLE
Issue 3 - As a guest I would like to see a calendar of events

### PR DESCRIPTION
**Pull request by jaysonbucy**
This was interesting. I updated the issue with the info but Google's calendar API doesn't allow for customizing displays. You _can_ do minimal editing in their embedding tool but you have to use that to display the calendar. I will use the API in a future issue to let the admins add a calendar event programmatically.

### CODE REVIEW

File: app/assets/config/client_id.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"installed":{"client_id":"852120292135-f784j6m8909uena2q8vgkifi10un298u.apps.googleusercontent.com","project_id":"boreal-physics-162914","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://accounts.google.com/o/oauth2/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"vKckKlgswyvZzK0E0UwUEu0A","redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]}}
```
Comment (kylessnell) [inline app/assets/config/client_id.json]: generally not a good idea to commit client ids and secrets. would recommend using environment variables instead. there's a handy gem called Figaro you can use to help with this.

File: app/views/welcome/_calendar.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<iframe src="https://calendar.google.com/calendar/embed?showTitle=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;showTz=0&amp;height=240&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=greateradaswimclub%40gmail.com&amp;color=%2329527A&amp;ctz=America%2FChicago"
+        style="border:solid 1px #39B348"
+        width="240"
+        height="240"
+        frameborder="0"
+        scrolling="no">
+</iframe>
```
Comment (kylessnell) [inline app/views/welcome/_calendar.html.erb:7]: why an iframe?

File: client_secret.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"installed":{"client_id":"852120292135-f784j6m8909uena2q8vgkifi10un298u.apps.googleusercontent.com","project_id":"boreal-physics-162914","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://accounts.google.com/o/oauth2/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"vKckKlgswyvZzK0E0UwUEu0A","redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]}}
```
Comment (kylessnell) [inline client_secret.json]: same comment as above

File: quickstart.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
+require 'google/apis/gmail_v1'
```
Comment (kylessnell) [inline quickstart.rb]: probably want this as an initializer, not living in the root of your project

File: quickstart.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+service = Google::Apis::GmailV1::GmailService.new
+service.client_options.application_name = APPLICATION_NAME
+service.authorization = authorize
+
+# Show the user's labels
+user_id = 'me'
+result = service.list_user_labels(user_id)
+
+puts "Labels:"
+puts "No labels found" if result.labels.empty?
```
Comment (kylessnell) [inline quickstart.rb]: want these?

File: quickstart.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  authorizer = Google::Auth::UserAuthorizer.new(
+    client_id, SCOPE, token_store)
+  user_id = 'default'
+  credentials = authorizer.get_credentials(user_id)
+  if credentials.nil?
+    url = authorizer.get_authorization_url(
+      base_url: OOB_URI)
+    puts "Open the following URL in the browser and enter the " +
+         "resulting code after authorization"
+    puts url
```
Comment (kylessnell) [inline quickstart.rb]: want these?

File: app/assets/config/client_id.json (hunk 1 above)
Comment (jaysonbucy) [reply app/assets/config/client_id.json]: agreed, I was required to do it by google to run their ruby program to tie my calendars into the program. I might put that on the .gitignore file though

File: app/views/welcome/_calendar.html.erb (hunk 2 above)
Comment (jaysonbucy) [reply app/views/welcome/_calendar.html.erb:7]: Google's embedding link they supply

File: quickstart.rb (hunk 4 above)
Comment (jaysonbucy) [reply quickstart.rb]: This is a google supplied program that you have to have. Literally they tell you to copy and paste it.

File: quickstart.rb (hunk 5 above)
Comment (jaysonbucy) [reply quickstart.rb]: same as above

File: quickstart.rb (hunk 6 above)
Comment (jaysonbucy) [reply quickstart.rb]: same as above

File: Gemfile [hunk 7]
```diff
@@ -8,6 +8,8 @@ gem 'uglifier', '>= 1.3.0'
 gem 'jquery-rails'
 gem 'turbolinks', '~> 5'
 gem 'bootstrap-sass'
+gem 'google-api-client', '>= 0.8'
+gem 'coffee-rails'
```
Comment (alexstophel) [inline Gemfile]: Where are you using this?

**Review by alexstophel — CHANGES_REQUESTED** [review]
Can you add documentation to the README for what someone needs to do locally to set up secrets for the project? I noticed there's a client_secret.json and quickstart.rb file in the .gitignore file. If I were to clone the repo fresh, what would I need to do?

Once you've done that, I'm good with the PR and you can merge it.

**Pull request closed without merging**